src/goto-symex/show_program.cpp

```diff
@@ -78,3 +84,243 @@ void show_program(const namespacet &ns, const symex_target_equationt &equation)
+{
+  switch(type)
+  {
+  case byte_op_type::BYTE_EXTRACT:
+    return "numOfExtracts";
+  case byte_op_type::BYTE_UPDATE:
+    return "numOfUpdates";
+  }
+  UNREACHABLE;
+}
+
+void show_byte_op_json(
+  std::ostream &out,
+  const namespacet &ns,
+  const symex_target_equationt &equation,
+  const byte_op_type type,
+  json_objectt &byte_ops_stats)
+{
+  // Get key values to be used in the json output based on byte operation type
+  // 1. json_get_key_byte_op_stats(byte_op_type):
+  //    returns relevant json object key string where object records
+  //    statistics for given byte operation.
+  // 2. json_get_key_byte_op_list(byte_op_type):
+  //    returns relevant json object key string where object records
+  //    a list of expressions for given byte operation.
+  // 3. json_get_key_byte_op_num(byte_op_type):
+  //    returns relevant json object key string where object number
+  //    of given byte operation.
+
+  const std::string key_byte_op_stats = json_get_key_byte_op_stats(type);
+  const std::string key_byte_op_list = json_get_key_byte_op_list(type);
+  const std::string key_byte_op_num = json_get_key_byte_op_num(type);
+
+  json_objectt &byte_op_stats = byte_ops_stats[key_byte_op_stats].make_object();
+  json_arrayt &byte_op_list = byte_op_stats[key_byte_op_list].make_array();
+
+  std::size_t equation_byte_op_count = 0;
+  for(const auto &step : equation.SSA_steps)
+  {
+    if(duplicated_previous_step(step))
+      continue;
+
+    const exprt &ssa_expr = step.get_ssa_expr();
+    const std::size_t ssa_expr_byte_op_count =
+      get_byte_op_count(type, ssa_expr);
+
+    if(ssa_expr_byte_op_count == 0)
+      continue;
+
+    equation_byte_op_count += ssa_expr_byte_op_count;
+    show_ssa_step_json(out, ns, step, ssa_expr, byte_op_list);
+  }
+
+  byte_op_stats[key_byte_op_num] =
+    json_numbert(std::to_string(equation_byte_op_count));
+}
+
+bool is_outfile_specified(const optionst &options)
+{
+  const std::string &filename = options.get_option("outfile");
+  return (!filename.empty() && filename != "-");
+}
+
+void show_byte_ops(
+  const optionst &options,
+  ui_message_handlert &ui_message_handler,
+  const namespacet &ns,
+  const symex_target_equationt &equation)
+{
+  const std::string &filename = options.get_option("outfile");
+  const bool outfile_given = is_outfile_specified(options);
+
+  std::ofstream of;
+
+  if(outfile_given)
+  {
+    of.open(filename, std::fstream::out);
+    if(!of)
+      throw invalid_command_line_argument_exceptiont(
+        "failed to open output file: " + filename, "--outfile");
+  }
+
+  std::ostream &out = outfile_given ? of : std::cout;
+
+  messaget msg(ui_message_handler);
+  switch(ui_message_handler.get_ui())
+  {
+  case ui_message_handlert::uit::XML_UI:
+    msg.error()
+      << "XML UI not supported for displaying byte extracts and updates."
+      << " Try --json-ui instead" << messaget::eom;
+    return;
+
+  case ui_message_handlert::uit::JSON_UI:
+  {
```

Comment:
:pick: I would factor out some of the blocks of code for these cases into a separate functions, in order to keep the size of the `show_byte_ops` function smaller.

Reply:
Comment addressed. Pushed changes.